KubeArmor/core/kubeArmor.go

```diff
@@ -694,22 +696,38 @@ func KubeArmor() {
 	// == //
 
 	if dm.K8sEnabled && cfg.GlobalCfg.Policy {
-		// watch k8s pods
-		go dm.WatchK8sPods()
-		dm.Logger.Print("Started to monitor Pod events")
 
 		// watch security policies
-		go dm.WatchSecurityPolicies()
+		securityPoliciesSynced := dm.WatchSecurityPolicies()
+		if securityPoliciesSynced == nil {
+			return
+		}
 		dm.Logger.Print("Started to monitor security policies")
 
 		// watch default posture
-		go dm.WatchDefaultPosture()
+		defaultPostureSynced := dm.WatchDefaultPosture()
+		if defaultPostureSynced == nil {
+			return
+		}
 		dm.Logger.Print("Started to monitor per-namespace default posture")
 
 		// watch kubearmor configmap
-		go dm.WatchConfigMap()
+		configMapSynced := dm.WatchConfigMap()
+		if configMapSynced == nil {
+			return
+		}
 		dm.Logger.Print("Watching for posture changes")
 
+		synced := cache.WaitForCacheSync(wait.NeverStop, securityPoliciesSynced, defaultPostureSynced, configMapSynced)
```

Comment:
I don't think it's a good idea to user `wait.NeverStop` here
It was okay to use `wait.NeverStop` earlier because they ran inside a go routine. But this is the main thread. Blocking code in a main thread could possibly result in the program to be stuck here not responding to anything. 
Maybe a context with certain timeout would help here?

Reply:
Is there a certain timeout value that I should use here? Or should I add a CLI arg/config entry for it and maybe go with a default 60s, and end-users can override it if necessary?

Reply:
I think it should be okay to keep a 60 second timeout. Sure let's have a flag we will reuse that timeout at other places when needed.